ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -343,16 +343,28 @@ orderby submission.TimeSubmitted
                 PuzzlesCausingGlobalLockout = await PuzzleStateHelper.PuzzlesCausingGlobalLockout(_context, Event, Team).ToListAsync();
             }
 
-            IQueryable<PuzzleUser> currentAuthorsQ = _context.PuzzleAuthors.Where(m => m.Puzzle == Puzzle).Select(m => m.Author);
-            List<PuzzleUser> CurrentAuthors = await currentAuthorsQ.OrderBy(p => p.Name).ToListAsync();
-            for (int i = 0; i < CurrentAuthors.Count; i++) {
-                if (CurrentAuthors[i].Name?.Trim().Length > 0)
+            if ((Puzzle.CustomAuthorText != null) && (Puzzle.CustomAuthorText.Trim().Length > 0))
+            {
+                PuzzleAuthor = "by " + Puzzle.CustomAuthorText.Trim();
+            }
+            else
+            {
+                IQueryable<PuzzleUser> currentAuthorsQ = _context.PuzzleAuthors.Where(m => m.Puzzle == Puzzle).Select(m => m.Author);
+                List<PuzzleUser> CurrentAuthors = await currentAuthorsQ.OrderBy(p => p.Name).ToListAsync();
+                for (int i = 0; i < CurrentAuthors.Count; i++)
                 {
-                    if ((PuzzleAuthor != null) && (PuzzleAuthor.Length > 0))
+                    if (CurrentAuthors[i].Name?.Trim().Length > 0)
                     {
-                        PuzzleAuthor += ", ";
+                        if ((PuzzleAuthor != null) && (PuzzleAuthor.Length > 0))
+                        {
+                            PuzzleAuthor += ", ";
+                        }
+                        else if ((PuzzleAuthor == null) || ((PuzzleAuthor != null) && (PuzzleAuthor.Length == 0)))

```

Comment:
The != null check is redundant since that's the only way to be on the right side of the ||.

This would be cleaner overall if you set `PuzzleAuthor = "";` on line 354 above the for loop. Then lines 358-365 boil down to this single line of readable code:

`PuzzleAuthor += (PuzzleAuthor.Length == 0) ? "by " : ", ";`

Reply:
I like it!